apps/mark-scan/frontend/src/pages/admin_screen.tsx

```diff
@@ -79,7 +77,9 @@ export function AdminScreen({
   if (!getPrecinctSelectionQuery.isSuccess) {
     return null;
   }
-  const precinctSelection = getPrecinctSelectionQuery.data.ok();
+
+  const precinctSelection: Optional<PrecinctSelection> =
+    getPrecinctSelectionQuery.data || undefined;
```

Comment:
This gets passed to `ElectionInfoBar` during render. That component's props allow `undefined` `PrecinctSelection` but not `null`. It's probably safe to update that component to accept `precinctSelection?: PrecinctSelection | null` but it seemed lower impact + very explicit to do it this way.

Reply:
Agreed.

I would even consider isolating the conversion between `null` and `undefined` to the query handler to keep it as localized as possible, since it's really a react-query quirk.

E.g. change the backend method to return `undefined`, then:
```ts
  useQuery() {
    const apiClient = useApiClient();
    return useQuery(
      this.queryKey(),
      // Since query functions are not allowed to return undefined, convert to null
      async () => (await apiClient.getPrecinctSelection()) ?? null,
      // Convert back to undefined when reading the query results
      { select: (precinctSelection) => precinctSelection ?? undefined }
    );
  },
```
I don't know if this will actually work.